api/internal/crawl/cmd/kustomize_stats/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"sigs.k8s.io/kustomize/api/internal/crawl/index"
+	"sigs.k8s.io/kustomize/api/konfig"
+)
+
+// iterateArr adds each item in arr into countMap.
+func iterateArr(arr []string, countMap map[string]int) {
+	for _, item := range arr {
+		if _, ok := countMap[item]; !ok {
+			countMap[item] = 1
+		} else {
+			countMap[item]++
+		}
+	}
+
+}
+
+// isKustomizationFile determines whether a file path is a kustomization file
+func isKustomizationFile(path string) bool {
+	basename := filepath.Base(path)
+	for _, name := range konfig.RecognizedKustomizationFileNames() {
+		if basename == name {
+			return true
+		}
+	}
+	return false
+}
+
+func main() {
+	topKindsPtr := flag.Int("kinds", -1,
+		"the number of kubernetes object kinds to be listed according to their popularities." +
+		"(By default, all the kinds will be listed. " +
+		"If you only want to list the 10 most popular kinds, set the flag to 10.)")
+	topIdentifiersPtr := flag.Int("identifiers", -1,
+		"the number of identifiers to be listed according to their popularities." +
+		"(By default, all the identifiers will be listed. " +
+		"If you only want to list the 10 most popular identifiers, set the flag to 10.)")
+	topKustomizeFeaturesPtr := flag.Int("kustomize-features", -1,
+		"the number of kustomize features to be listed according to their popularities." +
+			"(By default, all the features will be listed. " +
+			"If you only want to list the 10 most popular features, set the flag to 10.)")
+	flag.Parse()
+
+	ctx := context.Background()
+	idx, err := index.NewKustomizeIndex(ctx)
+	if err != nil {
+		fmt.Printf("Could not create an index: %v\n", err)
```

Comment:
log.Print("Could not..)
log.Fatal(err)


Presumably we'd want this to go to stderr, and want the process to exit with non-zero status

Reply:
Done